### PR TITLE
Tags

### PR DESCRIPTION
**Pull request by sldsmkd**
Override the troposphere Tags object so that KVPs are deduplicated - right side takes precedence; a = b(x='foo') + c(x='bar'); a == (x='bar')

Added Tags automation and inheritance; Tags defined at VPC level will be applied to all objects that support tags in the Template. Tags on child objects can be overridden.

There's some extra doc in the README to show how this works, and an example of working with tags in examples.

### CODE REVIEW

File: tropopause/ec2.py [hunk 1]
```diff
@@ -63,13 +88,15 @@ def __init__(self, title, template, **kwargs):
...
     ''' Overrides Subnet, routes traffic through an existing NAT gateway '''
+    @AddTagsFromVPC
     def __init__(self, title, template, **kwargs):
         # TODO: depends on public NAT
         super(PrivateSubnet, self).__init__(title, template, **kwargs)
         route_table = RouteTable(
             title + 'routetable',
             template,
-            VpcId=self.properties['VpcId']
+            VpcId=self.properties['VpcId'],
```
Comment (sldsmkd) [inline tropopause/ec2.py:98]: need to check whether Ref or not Ref is correct here

**Review by h-bragg — COMMENTED** [review]
Code looks good.

I think this is missing some specific tests that when you create a subnet or whatever it inherits the Tags from the VPC. It is only testing that it is an instance of Tags.

File: tropopause/ec2.py (hunk 1 above)
Comment (sldsmkd) [reply tropopause/ec2.py:98]: no need for Ref, removed from PublicSubnet

**Comment by sldsmkd** [comment]
Added extra test coverage in

File: README.md [hunk 2]
```diff
@@ -63,6 +63,14 @@ tropopause usage is similar to that of troposphere, the principle difference is
 
 ## Available Objects
 
+### tropopause
+* `Tags` - A replacement for troposphere Tags, concatenating tags deduplicates Keys, with precendence to the rightmost expression.
```
Comment (h-bragg) [inline README.md:67]: Markdown lint:

Should be a new line after header items.
Should be a new line before ``` blocks.

**Pull request merged**